discovery/gossiper.go

```diff
@@ -2924,6 +2945,18 @@ func (d *AuthenticatedGossiper) handleChanUpdate(nMsg *networkMsg,
 			routing.ErrVBarrierShuttingDown,
 		) {
 
+			log.Debugf("Update edge for short_chan_id(%v) got: %v",
+				shortChanID, err)
+		} else if strings.Contains(err.Error(),
+			channeldb.ErrEdgeNotFound.Error()) {
```

Comment:
You are right this is something I still not figured out (referring to the race condition here), I don't think its a DoS vector tho because the case where we do not find the corresponding edge is already covered here: https://github.com/lightningnetwork/lnd/blob/14a6f539717232c100e71447d7bcf7610c210559/discovery/gossiper.go#L2768

The problem why we need to check for this case here, is that somehow for Alias channels we remove and add the edge after six confirmations and we might run into the case where we are receiving the new ChanUpdate (with the __dont_foward__ bit unset) and also removing the edge leading to the case where the itests break. I guess I need to make sure that this call :
https://github.com/lightningnetwork/lnd/blob/14a6f539717232c100e71447d7bcf7610c210559/funding/manager.go#L3474

and the call here:
https://github.com/lightningnetwork/lnd/blob/14a6f539717232c100e71447d7bcf7610c210559/discovery/gossiper.go#L2941

do not interfere.

Reply:
This should be a separate commit with step-by-step info in the commit body. One TODO was to make the deletion + addition of edges atomic. That could be a separate PR that lands before this one.

Reply:
sounds good, I separated this commit and added a detailed commit body. I also included a TODO so that depending on whether this separate PR which makes the deletion+addition atomic lands before of after this change we can just delete it again. 
However I am also in favour of adding a separate PR before this one and removing this commit altogether because I don't think we need it if its atomic. I will create an issue or a PR directly.